jax/_src/distributed.py

```diff
@@ -131,6 +131,8 @@ def initialize(coordinator_address: Optional[str] = None,
       port does not matter, so long as the port is available on the coordinator
       and all processes agree on the port.
       May be ``None`` only on supported environments, in which case it will be chosen automatically.
+      Note that special addresses like ``localhost`` or ``127.0.0.1`` usually mean that the program
+      will bind to a local interface and are not suited when running in a multi-node environment.
```

Comment:
Let's call multi-node `multi-process`?

Reply:
Per offline discussion, we'll rename 'multi-hosts'. Multi-process but single-host should work with localhost and/or 127.0.0.1, but I will confirm that.

Reply:
I checked and on a single host, localhost and 127.0.0.1 both work fine.